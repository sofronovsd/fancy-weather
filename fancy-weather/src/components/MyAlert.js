import React from "react";
import {Alert} from "react-bootstrap";
import localesJson from "../utils/localesJson";

export default function MyAlert({alert, closeAlert, language}) {
    let message = localesJson[language][alert.text];
    if (alert.searchValue) {
        message += alert.searchValue;
    }
    return (
        <div>
            {alert.isShow ?
                <div className="alert-container">
                    <Alert data-testid="alert" variant="danger" onClose={closeAlert} dismissible>
                        {message}
                    </Alert>
                </div>
                : null}
        </div>
    );
}
