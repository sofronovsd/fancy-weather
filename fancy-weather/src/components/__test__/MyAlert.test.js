import MyAlert from "../MyAlert";
import React from "react";
import ReactDOM from "react-dom";
import {render} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom"

describe('Test alert functionality', () => {
  it('render alert without crashing', () => {
    const div = document.createElement('div');
    const alert = {isShow: true, text: "test"};
    ReactDOM.render(<MyAlert alert={alert} language="RU"/>, div)
  });

  it('render alert correctly', () => {
    const alert = {isShow: true, text: "test", searchValue: 'value'};
    const {getByTestId} = render(<MyAlert alert={alert} language="RU"/>);
    expect(getByTestId('alert')).toHaveStyle('display: block')
  });

  it('alert matches snapshot 1', () => {
    const alert = {isShow: true, text: "alertPlace", searchValue: 'City'};
    const tree = renderer.create(<MyAlert alert={alert} language="EN"/>);
    expect(tree).toMatchSnapshot();
  });

  it('alert matches snapshot 2', () => {
    const alert = {isShow: true, text: "alertPlace", searchValue: 'Город'};
    const tree = renderer.create(<MyAlert alert={alert} language="RU"/>);
    expect(tree).toMatchSnapshot();
  });

  it('render alert invisible', () => {
    const alert = {isShow: false, text: "test", searchValue: 'value'};
    render(<MyAlert alert={alert} language="RU"/>);
    expect(document.querySelector('.alert')).toEqual(null)
  });
});
