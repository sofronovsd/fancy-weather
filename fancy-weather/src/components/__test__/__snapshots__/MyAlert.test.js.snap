// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test alert functionality alert matches snapshot 1 1`] = `
<div>
  <div
    className="alert-container"
  >
    <div
      className="fade alert alert-danger alert-dismissible show"
      data-testid="alert"
      role="alert"
    >
      <button
        className="close"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only"
        >
          Close alert
        </span>
      </button>
      Can't find this place: City
    </div>
  </div>
</div>
`;

exports[`Test alert functionality alert matches snapshot 2 1`] = `
<div>
  <div
    className="alert-container"
  >
    <div
      className="fade alert alert-danger alert-dismissible show"
      data-testid="alert"
      role="alert"
    >
      <button
        className="close"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only"
        >
          Close alert
        </span>
      </button>
      Не удаётся найти это место: Город
    </div>
  </div>
</div>
`;
